### ISSUE TITLE
False positive E0308: static sized array type is icorrectly deduced instead of a dynamic slice

### ISSUE THREAD

**Issue by kryptan** [rust-lang/rust-analyzer#22113]
**rust-analyzer version**: rust-analyzer version: 0.3.2870-standalone (adef948679 2026-04-19)
**rustc version**: rustc 1.97.0-nightly (e22c616e4 2026-04-19)
**vscode extension**: 0.3.2870

**code snippet to reproduce**:
```rust
static S: &[()] =  &[];
```

this shows an error:
```
expected &[()], found &[(); 0] rust-analyzer (E0308)
```

This must be a recent regression as there was previously no error in this code.

This error only appears on Windows but not on Linux (WSL) even though versions of rust-analyzer & rustc are the same.

**Comment by ShoyuVanilla** [comment]
Can't reproduce. I guess you might be lacking `rust-src` component or having problem on loading sysroot on your Windows, which contains a `CoerceUnsized` trait that necessary for the unsized coercions.

**Comment by kryptan** [comment]
Thanks, rust-src was indeed missing.

**Comment by dvdplm** [comment]
Ran into the same issue (MacOS) and `rustup component add rust-src` is the fix. It wasn't very easy to connect the dots though and I wonder what we can do to make it clearer to users that the `rust-src` component is a requirement for rust-analyzer to work correctly.

**Comment by ChayimFriedman2** [comment]
It's already in the documentation and AFAIK the VSCode extension suggests to install it if you don't have it. I don't think we can do much more.
